src/Files.App/ViewModels/UserControls/Previews/ShellPreviewViewModel.cs

```diff
@@ -112,43 +104,71 @@ private nint WndProc(HWND hwnd, uint msg, nint wParam, nint lParam)
 
 				try
 				{
-					currentHandler = new PreviewHandler(clsid.Value, hwnd.DangerousGetHandle());
-					currentHandler.InitWithFileWithEveryWay(Item.ItemPath);
-					currentHandler.DoPreview();
+					_previewHandler = new PreviewHandler(clsid!.Value, hwnd);
+					_previewHandler.InitWithFileWithEveryWay(Item.ItemPath);
+					_previewHandler.DoPreview();
 				}
 				catch
 				{
 					UnloadPreview();
 				}
 			}
-			else if (msg == (uint)WindowMessage.WM_DESTROY)
+			else if (msg is PInvoke.WM_DESTROY)
 			{
-				if (currentHandler is not null)
+				if (_previewHandler is not null)
 				{
-					currentHandler.Dispose();
-					currentHandler = null;
+					_previewHandler.Dispose();
+					_previewHandler = null;
 				}
 			}
 
-			return DefWindowProc(hwnd, msg, wParam, lParam);
+			return PInvoke.DefWindowProc(hwnd, msg, wParam, lParam);
 		}
 
-		public void LoadPreview(UIElement presenter)
+		public unsafe void LoadPreview(UIElement presenter)
 		{
 			var parent = MainWindow.Instance.WindowHandle;
+			var hInst = PInvoke.GetModuleHandle(default(PWSTR));
+			var szClassName = $"{GetType().Name}-{Guid.NewGuid()}";
```

Comment:
```suggestion
			var szClassName = $"{nameof(ShellPreviewViewModel)}-{Guid.NewGuid()}";
```

Use the type name directly or renaming it to something more meaningful?

Reply:
Done. Id like to actually made a final decision when we finally finish the whole refactoring of this area.